core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -197,35 +197,27 @@ type cChainBodyExtras struct {
 
 var _ BodyHooks = (*cChainBodyExtras)(nil)
 
-func (e *cChainBodyExtras) AppendRLPFields(b rlp.EncoderBuffer, _ bool) error {
-	b.WriteUint64(uint64(e.Version))
-
-	var data []byte
-	if e.ExtData != nil {
-		data = *e.ExtData
-	}
-	b.WriteBytes(data)
-
-	return nil
+func (e *cChainBodyExtras) RLPFieldsForEncoding(b *Body) ([]any, []any) {
```

Comment:
named returns would help readability

```suggestion
func (e *cChainBodyExtras) RLPFieldsForEncoding(b *Body) (required, optional []any) {
```